dali/aux/optical_flow/turing_of/optical_flow_turing.cu

```diff
@@ -81,26 +93,61 @@ RgbToRgbaKernel(const uint8_t *input, uint8_t *output, size_t pitch, size_t widt
 }
 
 
+__global__ void
+BgrToRgbaKernel(const uint8_t *input, uint8_t *output, size_t pitch, size_t width_px,
+                size_t height) {
+  constexpr size_t in_channels = 3, out_channels = 4;
+  size_t x = threadIdx.x + blockIdx.x * blockDim.x;
+  size_t y = threadIdx.y + blockIdx.y * blockDim.y;
+  if (x >= width_px || y >= height) return;
+  size_t in_idx = in_channels * x + in_channels * width_px * y;
+  size_t out_idx = out_channels * x + pitch * y;
+  output[out_idx] = input[in_idx + 2];
```

Comment:
You should first read all 3 channels to local variables and only then store the result. Moreover, you should read using __ldg.

Reply:
Also, using vector types (uchar3/uchar4) might provide some benefit here - especially uchar4 for the output.

Reply:
A nitpick, but using vector types and a single load/store would be faster in unoptimized builds.